js/plugin/connection.ts

```diff
@@ -27,7 +28,30 @@ export const ConnectionMixin = (SuperClass) => {
       this.dispatchEvent(new CustomEvent(event, { detail, bubbles: true }));
     }
 
+    private onConnectionReady = () => {
+      this.connected = true;
+      this.fireEvent("browser-mod-connected");
+    }
+
+    private onConnectionLost = () => {
+      this.connected = false;
+      this.ready = false;
+      this.fireEvent("browser-mod-disconnected");
+    }
+
     private incoming_message(msg) {
+      // Check for connection state
+      if (!this.connected) {
+        this.onConnectionReady();
+      }
+      // Check for readiness
+      if (!this.ready) {
+        this.LOG("Integration ready: WebSocket connected and browser_mod loaded");
+        this.ready = true;
+        this.fireEvent("browser-mod-ready");
+        window.setTimeout(() => this.sendUpdate({}), 1000);
+      }
```

Comment:
`window.setTimeout(() => this.sendUpdate({}), 1000);` stays here because after re-connection not all entities were getting updated.

I think we should remove it in favor of resending this data on `browser-mod-ready` JS event.

Reply:
I agree to moving out to `browser-mod-ready` event.